subprojects/logging/src/main/java/org/gradle/internal/logging/console/WorkInProgressRenderer.java

```diff
@@ -103,87 +99,59 @@ private void resizeTo(int newBuildProgressLabelCount) {
         for (int i = newBuildProgressLabelCount - 1; i >= previousBuildProgressLabelCount; --i) {
             unusedProgressLabels.push(progressArea.getBuildProgressLabels().get(i));
         }
-
-        // Try to empty the unassigned progress operations
-        while (!unusedProgressLabels.isEmpty() && !unassignedProgressOperations.isEmpty()) {
-            attach(unassignedProgressOperations.pop());
-        }
     }
 
     private void attach(ProgressOperation operation) {
         // Skip attach if a children is already present
-        if (isChildAssociationAlreadyExists(operation.getOperationId())) {
+        if (registry.find(operation.getOperationId()).hasChildren()) {
```

Comment:
I think it won't be a problem, any operation with no children will be displayed. A child operation _pause_ the execution of a parent operation. If this assumption is wrong, then we will need to change the logic here.